### PR TITLE
[#2809]A judgment condition is added when gray rules match instances. This condition applies to scenarios where instances cannot be matched in cross-app invoking.

### PR DESCRIPTION
**Pull request by m13922236379**
A judgment condition is added when gray rules match instances. This condition applies to scenarios where instances cannot be matched in cross-app invoking.
From:#2809

### CODE REVIEW

File: governance/src/main/java/org/apache/servicecomb/router/distribute/AbstractRouterDistributor.java [hunk 1]
```diff
@@ -146,7 +150,7 @@ public void initLatestVersion(String serviceName, List<T> list) {
     String latestVersion = null;
     for (T server : list) {
       E ms = getIns.apply(server);
-      if (getServerName.apply(ms).equals(serviceName)) {
+      if (getServerName.apply(ms).equals(serviceName) || getAlias.apply(ms).equals(serviceName)) {
         if (latestVersion == null || VersionCompareUtil
```
Comment (zhaozw) [inline governance/src/main/java/org/apache/servicecomb/router/distribute/AbstractRouterDistributor.java:154]: getAlias如何可以为null，加上getAlias!=null&&getAlias.apply(ms).equals(serviceName) ?

Comment (m13922236379) [reply governance/src/main/java/org/apache/servicecomb/router/distribute/AbstractRouterDistributor.java:154]: 同理，那getServerName应该也不能为null吧。

**Comment by liubao68** [comment]
I'd prefer to keep  targetServiceName parameter

```
 public <T, E> List<T> getFilteredListOfServers(List<T> list,
                                                 String targetServiceName, Map<String, String> headers, RouterDistributor<T, E> distributer) {
```

in RouterFilter will always be microservice name (do not contains application name).

**Comment by m13922236379** [comment]
Fix the problem by fixing targetServiceName to the microservice name.

**Pull request closed without merging**